src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -168,5 +167,17 @@ void TryRemoveSlow(IEnumerable c)
                     TryRemove(o);
             }
         }
+
+        private static ArrayList ToArrayList(IEnumerable items)
+        {
+            if (items is ICollection ic)
```

Comment:
minor optimization to avoid O(n) capacity checks and potential resizing of the ArrayList. Two common collection types (arrays and list<T>) implement this

Reply:
This has broken our custom collection type: https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate/Collection/Generic/SetHelpers/SetSnapShot.cs

https://github.com/nhibernate/nhibernate-core/pull/3088

Reply:
Sorry about that @hazzik 
I think a fix for this was done on the 3.13.x branch in https://github.com/nunit/nunit/pull/4097, but it hasn't been released yet.
To be sure, would you mind opening an issue with the details specific to what you're hitting? At a minimum, perhaps it could begin a conversation about timelines for the next 3.13.x that

Reply:
@stevenaw, ok, sounds good. I think #4097 covers it. Thanks